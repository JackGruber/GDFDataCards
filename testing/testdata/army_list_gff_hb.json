{
    "gameSystem": "gff",
    "armyId": "7o6om21wxlvvy3hq",
    "armyIds": [
        "7o6om21wxlvvy3hq"
    ],
    "armyFaction": "Havoc Brothers",
    "armyName": "Havoc Brothers",
    "modified": "2023-05-29T12:45:16.200Z",
    "saveVersion": 3,
    "listPoints": 300,
    "list": {
        "id": "Px5Sfdtf",
        "key": "01vnHI6QWHzg5yPEps9P8",
        "name": "Havoc Brothers",
        "units": [
            {
                "id": "eD_a0Ea",
                "armyId": "7o6om21wxlvvy3hq",
                "selectionId": "LijrQ",
                "selectedUpgrades": [
                    {
                        "instanceId": "J8Z84gqyG",
                        "upgradeId": "R3s-sc-",
                        "optionId": "q5BXAEy"
                    },
                    {
                        "instanceId": "2BydmJ5Mo",
                        "upgradeId": "-2GgkX2",
                        "optionId": "qIHT5"
                    },
                    {
                        "instanceId": "9qhvwCU2d",
                        "upgradeId": "0tRGmMk",
                        "optionId": "36K0S"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "xikokQS",
                "armyId": "7o6om21wxlvvy3hq",
                "selectionId": "CxnJo",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "xikokQS",
                "armyId": "7o6om21wxlvvy3hq",
                "selectionId": "M-xYD",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "xikokQS",
                "armyId": "7o6om21wxlvvy3hq",
                "selectionId": "YzC2r",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "jINKf6R",
                "armyId": "7o6om21wxlvvy3hq",
                "selectionId": "HnWIF",
                "selectedUpgrades": [
                    {
                        "instanceId": "wchzl1pY3",
                        "upgradeId": "JxBta2M",
                        "optionId": "QvzQzfr"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "1qSAYjr",
                "armyId": "7o6om21wxlvvy3hq",
                "selectionId": "O6T3E",
                "selectedUpgrades": [
                    {
                        "instanceId": "lHHC6M0eG",
                        "upgradeId": "si4nIKu",
                        "optionId": "_U8-2MK"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            }
        ],
        "points": 300,
        "pointsLimit": 300,
        "creationTime": "1685364025629",
        "campaignMode": false,
        "competitive": true,
        "gameSystem": "gff",
        "selectedUnitId": "LijrQ",
        "unitPreview": null,
        "undoUnitRemove": [
            {
                "id": "jINKf6R",
                "cost": 65,
                "name": "Havoc Support",
                "size": 1,
                "defense": 3,
                "quality": 3,
                "upgrades": [
                    "F1"
                ],
                "equipment": [
                    {
                        "id": "TLV3p",
                        "name": "CCW",
                        "label": "CCW",
                        "attacks": 1,
                        "specialRules": [],
                        "count": 1,
                        "originalCount": 1,
                        "type": "ArmyBookWeapon"
                    },
                    {
                        "id": "y5U0c",
                        "label": "Heavy Machinegun",
                        "range": 30,
                        "attacks": 3,
                        "specialRules": [
                            {
                                "key": "ap",
                                "name": "AP",
                                "rating": "1",
                                "modify": false
                            }
                        ],
                        "name": "Heavy Machinegun",
                        "count": 1,
                        "originalCount": 1,
                        "type": "ArmyBookWeapon"
                    }
                ],
                "specialRules": [
                    {
                        "key": "relentless",
                        "name": "Relentless",
                        "rating": ""
                    }
                ],
                "disabledUpgradeSections": [],
                "armyId": "7o6om21wxlvvy3hq",
                "selectedUpgrades": [],
                "sortId": 6,
                "loadout": [
                    {
                        "id": "TLV3p",
                        "name": "CCW",
                        "label": "CCW",
                        "attacks": 1,
                        "specialRules": [],
                        "count": 1,
                        "originalCount": 1,
                        "type": "ArmyBookWeapon"
                    },
                    {
                        "id": "y5U0c",
                        "label": "Heavy Machinegun",
                        "range": 30,
                        "attacks": 3,
                        "specialRules": [
                            {
                                "key": "ap",
                                "name": "AP",
                                "rating": "1",
                                "modify": false
                            }
                        ],
                        "name": "Heavy Machinegun",
                        "count": 1,
                        "originalCount": 1,
                        "type": "ArmyBookWeapon"
                    }
                ],
                "selectionId": "NuTT1",
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            }
        ]
    },
    "favourite": false
}