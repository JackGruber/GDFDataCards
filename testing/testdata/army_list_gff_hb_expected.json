{
    "armyId": "7o6om21wxlvvy3hq",
    "gameSystem": "gff",
    "gameSystemId": 3,
    "armyName": "Havoc Brothers",
    "listPoints": 300,
    "listName": "Havoc Brothers",
    "units": [
        {
            "type": "Havoc Champion",
            "name": "Havoc Champion",
            "armyId": "7o6om21wxlvvy3hq",
            "id": "eD_a0Ea",
            "cost": 50,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "A1",
                "B1",
                "C1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 2,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Plasma Pistol",
                    "range": 12,
                    "specialRules": [],
                    "ap": "4"
                }
            ],
            "specialRules": [
                {
                    "key": "hero",
                    "name": "Hero",
                    "rating": "",
                    "label": "Hero"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "3",
                    "label": "Tough(3)"
                }
            ],
            "upgradeCost": [
                35,
                0,
                5
            ],
            "equipment": [
                {
                    "name": "Havoc Lord",
                    "specialRules": [
                        {
                            "key": "dark-tactics",
                            "name": "Dark Tactics",
                            "type": "ArmyBookRule",
                            "label": "Dark Tactics",
                            "rating": ""
                        }
                    ]
                }
            ]
        },
        {
            "type": "Havoc Brother",
            "name": "Havoc Brother",
            "armyId": "7o6om21wxlvvy3hq",
            "id": "xikokQS",
            "cost": 25,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "A1",
                "G1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Heavy Rifle",
                    "range": 24,
                    "specialRules": [],
                    "ap": "1"
                }
            ],
            "specialRules": []
        },
        {
            "type": "Havoc Brother",
            "name": "Havoc Brother",
            "armyId": "7o6om21wxlvvy3hq",
            "id": "xikokQS",
            "cost": 25,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "A1",
                "G1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Heavy Rifle",
                    "range": 24,
                    "specialRules": [],
                    "ap": "1"
                }
            ],
            "specialRules": []
        },
        {
            "type": "Havoc Brother",
            "name": "Havoc Brother",
            "armyId": "7o6om21wxlvvy3hq",
            "id": "xikokQS",
            "cost": 25,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "A1",
                "G1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Heavy Rifle",
                    "range": 24,
                    "specialRules": [],
                    "ap": "1"
                }
            ],
            "specialRules": []
        },
        {
            "type": "Havoc Support",
            "name": "Havoc Support",
            "armyId": "7o6om21wxlvvy3hq",
            "id": "jINKf6R",
            "cost": 65,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "F1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 3,
                    "count": 1,
                    "name": "Autocannon",
                    "range": 36,
                    "specialRules": [],
                    "ap": "2"
                }
            ],
            "specialRules": [
                {
                    "key": "relentless",
                    "name": "Relentless",
                    "rating": "",
                    "label": "Relentless"
                }
            ],
            "upgradeCost": [
                25
            ]
        },
        {
            "type": "Talon Raptor",
            "name": "Talon Raptor",
            "armyId": "7o6om21wxlvvy3hq",
            "id": "1qSAYjr",
            "cost": 35,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "A1",
                "I1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 4,
                    "count": 1,
                    "name": "Dual Energy Claws",
                    "specialRules": [
                        {
                            "key": "rending",
                            "name": "Rending",
                            "type": "ArmyBookRule",
                            "label": "Rending",
                            "rating": ""
                        }
                    ]
                }
            ],
            "specialRules": [
                {
                    "key": "ambush",
                    "name": "Ambush",
                    "rating": "",
                    "label": "Ambush"
                },
                {
                    "key": "furious",
                    "name": "Furious",
                    "rating": "",
                    "label": "Furious"
                },
                {
                    "key": "flying",
                    "name": "Flying",
                    "rating": "",
                    "label": "Flying"
                }
            ],
            "upgradeCost": [
                10
            ]
        }
    ]
}